deepmedic/logging/accuracyMonitor.py

```diff
@@ -146,12 +148,91 @@ def reportAccuracyForLastSubepoch(self) :
+        currSubep = self.numberOfSubepochsForWhichUpdated - 1
+
+        self.log.print3('=============== LOGGING TO TENSORBOARD ===============')
+        self.log.print3('Logging ' + trainOrValString + ' metrics')
+        self.log.print3('Epoch: ' + str(self.epoch) +
+                        ' | Subepoch ' + str(currSubep) + '/' + str(self.numberOfSubepochsPerEpoch - 1))
+        step_num = currSubep + (self.epoch * (self.numberOfSubepochsPerEpoch))
+        self.log.print3('Step number: ' + str(step_num))
+
+        # check if user included tensorboard logging in the config
+        if self.tensorboard_logger is None:
+            self.log.print3('Tensorboard logging not activated. Skipping...')
+            self.log.print3('======================================================')
+            return
+
+        # During training, also report the mean value of the Cost Function:
+        if self.training0orValidation1 == 0 and self.tensorboard_logger is not None:
+            self.log.print3('--- Logging average metrics for all classes ---')
+
+            # create metrics dictionary
+            metrics_dict = {'mean_acc': self.meanEmpiricalAccuracyOfEachSubep[currSubep],
+                            'mean_cost': self.meanCostOfEachSubep[currSubep]}
+            class_string = 'Class-all'
+            self.log_to_tensorboard(metrics_dict, class_string, step_num)
+
+            self.log.print3('Logged metrics: ' + str(list(metrics_dict.keys())))
+
+        # Report accuracy over subepoch for each class_i:
+        self.log.print3('--- Logging per class metrics ---')
+        for class_i in range(self.numberOfClasses):
+            classString = "Class-" + str(class_i)
+
+            [meanAccClassOfSubep,
+             meanAccOnPosOfSubep,
+             meanPrecOfSubep,
+             meanAccOnNegOfSubep,
+             meanDiceOfSubep] = self.listPerSubepPerClassMeanAccSensSpecDsc[currSubep][class_i] if class_i != 0 else \
+                self.listPerSubepForegrMeanAccSensSpecDsc[currSubep]  # If class-0, report foreground.
+
+            # create metrics dictionary
+            metrics_dict = {'mean_acc': meanAccClassOfSubep,
+                            'mean_sens': meanAccOnPosOfSubep,
+                            'mean_prec': meanPrecOfSubep,
+                            'mean_spec': meanAccOnNegOfSubep,
+                            'mean_dice': meanDiceOfSubep}
+            class_string = classString
+            self.log_to_tensorboard(metrics_dict, class_string, step_num)
+
+        self.log.print3('Logged metrics: ' + str(list(metrics_dict.keys())))
+        self.log.print3('======================================================')
+
+    def reportDSCWholeSegmentation(self, metrics_dict_list):
+        if self.tensorboard_logger is not None:
+            self.log.print3('=============== LOGGING TO TENSORBOARD ===============')
+            self.log.print3('Logging whole brain segmentation metrics')
```

Comment:
It is not brain segmentation specific

```suggestion
            self.log.print3('Logging validation metrics from segmentation of whole validation volumes.')
```